### PR TITLE
add development & CI test profiles

### PR DESCRIPTION
**Pull request by andreeaflorescu**
Added two test profiles: one for development (devel) and one for
running the integration tests as part of the CI.

Currently the only difference between the two is running the
coverage test. Specifically, in the `ci` profile the test fails
if current_coverage > previous_coverage, while in the `devel`
profile the coverage file is updated with the new value.

Signed-off-by: Andreea Florescu <fandree@amazon.com>

### CODE REVIEW

File: tests/test_coverage.py [hunk 1]
```diff
@@ -82,15 +83,25 @@ def _update_coverage(cov_value):
...
+            assert False, "Coverage is increased from {} to {}. " \
+                          "Please update the coverage in " \
+                          "tests/coverage.".format(
+                previous_coverage,
+                current_coverage
+            )
+        elif profile == pytest.profile_devel:
+            _update_coverage(current_coverage)
+        else:
+            # This should never happen because we pytest should only accept
```
Comment (alxiord) [inline tests/test_coverage.py]:
```suggestion
            # This should never happen because pytest should only accept
```

File: tests/test_coverage.py [hunk 2]
```diff
@@ -82,15 +83,25 @@ def _update_coverage(cov_value):
     with open(coverage_path, "w") as f:
         f.write(str(cov_value))
```
Comment (alxiord) [inline tests/test_coverage.py:85]: nit: can you add 2 blank lines between all top level functions while you're at it?

File: README.md [hunk 3]
```diff
@@ -32,10 +33,23 @@ tests/test_coverage.py::test_coverage PASSED                             [100%]
...
 ```
 
+### Test Profiles
+
+The integration tests support two test profiles:
+- **devel**: this is the recommended profile for running the integration tests on
+  a local development machine.
+- **ci** (default option): this is the profile used when running the integration tests as part of
+  the the Continuous Integration (CI).
+
```
Comment (sameo) [inline README.md:31]: Could we describe the main differences between the 2 profiles?

File: README.md [hunk 4]
```diff
@@ -5,6 +5,7 @@ TODO
 ## Running the tests
 
 ### Dependencies
+
```
Comment (sameo) [inline README.md]: Nit: Unrelated cleanup.

File: README.md (hunk 3 above)
Comment (andreeaflorescu) [reply README.md:31]: For now the only difference is in running the coverage test. This difference is described in the Adaptive Coverage section of the doc. For this particular crate I don't believe we will need anything else, but for the virtio-devices we might want to do performance testing and we probably cannot expect to have the same thresholds on a local machine vs on a i3.metal. We can use the same model for these crates in the future.

I have no good idea about describing the difference between the profiles. Should I just say that the coverage test behaves differently with different profiles? Do you have any better suggestions?

File: tests/test_coverage.py (hunk 1 above)
Comment (andreeaflorescu) [reply tests/test_coverage.py]: Done.

File: tests/test_coverage.py (hunk 2 above)
Comment (andreeaflorescu) [reply tests/test_coverage.py:85]: I'll add it in a future cleanup of the code.

File: README.md (hunk 3 above)
Comment (andreeaflorescu) [reply README.md:31]: Done.

**Comment by andreeaflorescu** [comment]
@aghecenco @sameo please re-review.

**Review by alxiord — APPROVED** [review]

**Review by sameo — APPROVED** [review]
